lib/customer.rb

```diff
@@ -0,0 +1,54 @@
+require_relative 'restaurant'
+
+class Customer
+  attr_reader :name, :order, :restaurant
+
+  def initialize(restaurant = Restaurant.new)
+    @restaurant = restaurant
+    @order = Hash.new(0)
+  end
+
+  def menu
+    restaurant.view_menu
+  end
+
+  def place_order(dish, quantity = 1)
+    raise "Please select a dish from the menu." if !restaurant.menu_include?(dish)
+    order[dish] += quantity
+    return "Thank you for ordering."
+  end
+
+  def review_order
+    raise "You haven't ordered anything." if order.empty?
+    order_summary
+  end
+
+  def check_out
+    fail 'Please make an order' if order.empty?
+    return "Total costs: £#{final_bill}."
+  end
+
+  def submit_order
+    restaurant.take_away
+  end
+
+  private
+
+  def order_summary
+    order.map { |dish, quantity| "#{dish}: #{quantity}"}.join(", ")
+  end
+
+  def total_cost_per_dish
+    order.map { |dish, quantity| "#{quantity} x #{dish}: £#{'%.2f' % (quantity * Restaurant::MENU[dish])}"}.join(", ")
```

Comment:
Metrics/LineLength: Line is too long. [118/80]<br>Style/FormatString: Favor `format` over `String#%`.